src/core/lib/security/credentials/tls/grpc_tls_certificate_provider.cc

```diff
@@ -202,10 +203,15 @@ void FileWatcherCertificateProvider::ForceUpdate() {
       (pem_key_cert_pairs.has_value() &&
        pem_key_cert_pairs_ != *pem_key_cert_pairs);
   if (identity_cert_changed) {
-    if (pem_key_cert_pairs.has_value()) {
-      pem_key_cert_pairs_ = std::move(*pem_key_cert_pairs);
+    absl::Status key_cert_match = PrivateKeyAndCertificateMatch(primary_key_path_, identity_certificate_path_);
+    if (key_cert_match.ok()){
+      if (pem_key_cert_pairs.has_value()) {
+        pem_key_cert_pairs_ = std::move(*pem_key_cert_pairs);
+      } else {
+        pem_key_cert_pairs_ = {};
+      }
     } else {
-      pem_key_cert_pairs_ = {};
+      identity_cert_changed = false;
```

Comment:
Add a comment above explaining why "identity_cert_changed = false" is logically equivalent to "key-cert match failed". Also we will need some loggings. You can use `gpr_log(GPR_ERROR, ....)`

Reply:
Ok, will add more comments. Why do we need the loggings?

Reply:
The key-cert mismatch basically is something unexpected, and we'd like to at least log this behavior when that happens. That may help debug certain errors while debugging.